azurerm/resource_arm_connection_monitor.go

```diff
@@ -0,0 +1,336 @@
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	watcherName := id.Path["networkWatchers"]
+	name := id.Path["connectionMonitors"]
+
+	resp, err := client.Get(ctx, resourceGroup, watcherName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Connection Monitor %q (Watcher %q / Resource Group %q) %+v", name, watcherName, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("network_watcher_name", watcherName)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if props := resp.ConnectionMonitorResultProperties; props != nil {
+		d.Set("auto_start", props.AutoStart)
+		d.Set("interval_in_seconds", int(*props.MonitoringIntervalInSeconds))
+
+		source := flattenArmConnectionMonitorSource(props.Source)
+		if err := d.Set("source", source); err != nil {
+			return fmt.Errorf("Error setting `source`: %+v", err)
+		}
+
+		dest := flattenArmConnectionMonitorDestination(props.Destination)
+		if err := d.Set("destination", dest); err != nil {
+			return fmt.Errorf("Error setting `destination`: %+v", err)
+		}
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmConnectionMonitorDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).connectionMonitorsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	watcherName := id.Path["networkWatchers"]
+	name := id.Path["connectionMonitors"]
+
+	future, err := client.Delete(ctx, resourceGroup, watcherName, name)
+	if err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error deleting Connection Monitor %q (Watcher %q / Resource Group %q): %+v", name, watcherName, resourceGroup, err)
+		}
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for the deletion of Connection Monitor %q (Watcher %q / Resource Group %q): %+v", name, watcherName, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func flattenArmConnectionMonitorSource(input *network.ConnectionMonitorSource) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	output := make(map[string]interface{})
+
+	if resourceID := input.ResourceID; resourceID != nil {
+		output["virtual_machine_id"] = *resourceID
+	}
+	if port := input.Port; port != nil {
+		output["port"] = *port
+	}
+
+	return []interface{}{output}
+}
+
+func expandArmConnectionMonitorSource(d *schema.ResourceData) (*network.ConnectionMonitorSource, error) {
+	sources := d.Get("source").([]interface{})
+	if len(sources) == 0 {
+		return nil, fmt.Errorf("Error expanding `source`: not found")
```

Comment:
As source is a required field this error will never happen?

Reply:
Yes. I think you're right. Same with the `destination` block.